libtransmission/peer-mgr.h

```diff
@@ -76,30 +83,51 @@ class tr_peer_info
 
     ~tr_peer_info()
     {
-        [[maybe_unused]] auto const n_prev = n_known_peers--;
-        TR_ASSERT(n_prev > 0U);
+        if (!std::empty(listen_socket_address_.port()))
+        {
+            [[maybe_unused]] auto const n_prev = n_known_connectable_--;
+            TR_ASSERT(n_prev > 0U);
+        }
     }
 
-    [[nodiscard]] static auto known_peer_count() noexcept
+    [[nodiscard]] static auto known_connectable_count() noexcept
```

Comment:
I think this can be `constexpr` now that `n_known_*` is no longer atomic

Reply:
`constexpr` is not valid here.

```
/home/test/git_src/Transmission/libtransmission/../libtransmission/peer-mgr.h:93:41: error: constexpr function never produces a constant expression [-Winvalid-constexpr]
    [[nodiscard]] constexpr static auto known_connectable_count() noexcept
                                        ^
/home/test/git_src/Transmission/libtransmission/../libtransmission/peer-mgr.h:95:16: note: read of non-const variable 'n_known_connectable_' is not allowed in a constant expression
        return n_known_connectable_;
               ^
/home/test/git_src/Transmission/libtransmission/../libtransmission/peer-mgr.h:490:24: note: declared here
    static auto inline n_known_connectable_ = size_t{};
```